src/config_generator/from_proto.rs

```diff
@@ -0,0 +1,387 @@
+#![allow(unused)]
+
+use std::collections::{BTreeSet, HashMap};
+
+use convert_case::{Case, Casing};
+use derive_setters::Setters;
+use prost_reflect::prost_types::{
+    DescriptorProto, EnumDescriptorProto, FileDescriptorSet, ServiceDescriptorProto,
+};
+use strum_macros::Display;
+
+use crate::blueprint::GrpcMethod;
+use crate::config::{Arg, Config, Field, Grpc, Tag, Type};
+
+pub(super) static DEFAULT_SPECTATOR: &str = "_";
+
+/// Enum to represent the type of the descriptor
+#[derive(Display, Clone)]
+pub enum DescriptorType {
+    Enum,
+    Message,
+    Query,
+}
+
+#[derive(Default)]
+struct Helper {
+    map: HashMap<String, String>,
+    package: String,
+}
+
+impl Helper {
+    fn get_value(&self, name: &str, ty: DescriptorType) -> String {
+        let package = self.package.replace('.', DEFAULT_SPECTATOR).to_uppercase();
+        match ty {
+            DescriptorType::Enum => {
+                format!("{}{}{}", package, DEFAULT_SPECTATOR, name)
+            }
+            DescriptorType::Message => {
+                format!("{}{}{}", package, DEFAULT_SPECTATOR, name)
+            }
+            DescriptorType::Query => format!(
+                "{}{}{}",
+                package.to_case(Case::Snake),
+                DEFAULT_SPECTATOR,
+                name.to_case(Case::Camel),
+            ),
+        }
+    }
+    fn insert(&mut self, name: &str, ty: DescriptorType) {
+        self.map.insert(
+            format!("{}.{}", self.package, name),
+            self.get_value(name, ty),
+        );
+    }
+    fn get(&self, name: &str) -> Option<String> {
+        self.map.get(&format!("{}.{}", self.package, name)).cloned()
+    }
+}
+
+fn convert_ty(proto_ty: &str) -> String {
```

Comment:
@coderabbitai add docs to all the functions in this file explaining what they do. Create it as a patch.

Reply:
go for it.